MailContainerTest/Services/MailTransferService.cs

```diff
@@ -1,93 +1,77 @@
+            return result;
+        }
+
+        private bool CanPerformTransferBetweenContainers(MailContainer sourceContainer, MailContainer destinationContainer)
+        {
+            bool canTransfer = true;
 
-            switch (request.MailType)
+            if (sourceContainer.Status != MailContainerStatus.Operational || destinationContainer.Status != MailContainerStatus.Operational)
             {
-                case MailType.StandardLetter:
-                    if (mailContainer == null)
-                    {
-                        result.Success = false;
-                    }
-                    else if (!mailContainer.AllowedMailType.HasFlag(AllowedMailType.StandardLetter))
-                    {
-                        result.Success = false;
-                    }
-                    break;
-
-                case MailType.LargeLetter:
-                    if (mailContainer == null)
-                    {
-                        result.Success = false;
-                    }
-                    else if (!mailContainer.AllowedMailType.HasFlag(AllowedMailType.LargeLetter))
-                    {
-                        result.Success = false;
-                    }
-                    else if (mailContainer.Capacity < request.NumberOfMailItems)
-                    {
-                        result.Success = false;
-                    }
-                    break;
-
-                case MailType.SmallParcel:
-                    if (mailContainer == null)
-                    {
-                        result.Success = false;
-                    }
-                    else if (!mailContainer.AllowedMailType.HasFlag(AllowedMailType.SmallParcel))
-                    {
-                        result.Success = false;
-
-                    }
-                    else if (mailContainer.Status != MailContainerStatus.Operational)
-                    {
-                        result.Success = false;
-                    }
-                    break;
+                canTransfer = false;
             }
 
-            if (result.Success)
+            if (sourceContainer.AllowedMailType != destinationContainer.AllowedMailType)
             {
-                mailContainer.Capacity -= request.NumberOfMailItems;
+                canTransfer = false;
+            }
 
-                if (dataStoreType == "Backup")
-                {
-                    var mailContainerDataStore = new BackupMailContainerDataStore();
-                    mailContainerDataStore.UpdateMailContainer(mailContainer);
+            return canTransfer;
+        }
 
-                }
-                else
-                {
-                    var mailContainerDataStore = new MailContainerDataStore();
-                    mailContainerDataStore.UpdateMailContainer(mailContainer);
-                }
-            }
+        private bool PerformTransfer(int numberOfItems, MailContainer sourceContainer, MailContainer destinationContainer)
+        {
```

Comment:
Error handling would be good here if I had more time, to ensure that either both updates are made, or neither